main.js

```diff
@@ -1,70 +1,98 @@
-  // create list li elemnts
-  const li = document.createElement('li');
-  li.className = `${book.title}${book.author}`;
-  const pForTitle = document.createElement('p');
-  const pForAuthor = document.createElement('p');
-  const pForDelete = document.createElement('button');
-  const hr = document.createElement('hr');
-  pForDelete.className = 'delete';
-  // add content to the elements
-  pForTitle.textContent = book.title;
-  pForAuthor.textContent = book.author;
-  pForDelete.textContent = 'Delete';
-  // add element to the list and the list to the ul
-  li.appendChild(pForTitle);
-  li.appendChild(pForAuthor);
-  li.appendChild(pForDelete);
-  li.appendChild(hr);
-  ul.appendChild(li);
-}
 // create the book list
 let books = [];
+
+// books count
+let bookCount = 0;
+
+// the book class
+const Book = {
+  title: '',
+  author: '',
+
+  // to creat the title and the author
+  setBook: function setbook(bookeTitle, bookeAuthor) {
+    this.title = bookeTitle;
+    this.author = bookeAuthor;
+  },
+
+  // to create and add the book
+  creatAndAddBooks: function addbook(ul) {
+    // create list li elemnts
+    const li = document.createElement('li');
+    li.className = `${this.title}${this.author}`;
+    const pForTitle = document.createElement('p');
+    const ulPWreaper = document.createElement('ul');
+    ulPWreaper.className = 'ulWrapper';
+    const pForAuthor = document.createElement('p');
+    const pForDelete = document.createElement('button');
+    pForDelete.className = 'delete';
+    // add content to the elements
+    pForTitle.textContent = this.title;
+    pForAuthor.textContent = this.author;
+    pForDelete.textContent = 'Remove';
+    li.innerHTML = `"${this.title}" by ${this.author} <button class="delete">Remove</button>`;
+    if (bookCount % 2 === 1) {
+      li.id = 'bg-color';
+    }
+    ul.appendChild(li);
+    return this;
+  },
+
+  deleteBook: function deletebook(e) {
+    if (e.target.className === 'delete') {
+      const li = e.target.parentElement;
+      li.parentNode.removeChild(li);
+      for (let i = 0; i < books.length; i += 1) {
+        let ClassNAme = books[i].title;
+        ClassNAme += books[i].author;
+        if (e.target.parentElement.className === ClassNAme) {
+          books.splice(i, 1);
+        }
+      }
+    }
+  },
+};
+let newBook = Book;
 if (localStorage.books) {
   books = JSON.parse(localStorage.books);
 }
+// instance new_book
+function clonebook(book) {
+  const clone = {};
+  Object.entries(book).forEach(([key, value]) => {
+    clone[key] = value;
+  });
+  return clone;
+}
 // call nedded elements from the document
 const bookList = document.querySelector('.book-list');
 const form = document.getElementById('abbBookForm');
 const title = document.getElementById('title');
 const author = document.getElementById('author');
+
 // create the books, add them the book list and local storage
 form.addEventListener('submit', (e) => {
   e.preventDefault();
```

Comment:
- [ ] One of the requirements says that you should refactor your code using classes, which means all logic regarding the book logic like store, adding a new book, removing a book.... should be methods inside your class. for example
```javascript
class Books {
  constructor() {
    this.books = [];
  }

  addBook(title, author) {
    id = generateID(); 
    this.books = this.books.concat({ title, author, author });
    return this.books; 
  }

  removeBook(bookID) {
    // bookID can be any other property, e.g. title or author, in as much is it unique
    this.books = this.books.filter(({ id }) => bookID !== id); // This assumes that `bookID` parameter is the book id
    return this.books; 
  }

saveBookInLocalstorage(book){
     .........
}

  // getter method
  getBooks() {
    return this.books;
  }
}

```

- [ ] Kindly work also on this error when in the store there are books with the same title, deleting one also deletes the other of the same title and is not removed in the browser. Kindly check this :point_down:  screenvideo for more clarification

https://user-images.githubusercontent.com/65068771/154213508-d813b63d-7943-47de-b797-250631366657.mp4